File: examples/basic_pool.rs
use rust_thread_pool_runtime::Runtime;

fn main() {
    let runtime = Runtime::new(4);

    let handle = runtime.spawn(|| {
        let mut sum = 0u64;
        for i in 0..1_000_000u64 {
            sum += i;
        }
        sum
    });

    let result = handle.join().unwrap();
    println!("result = {result}");

    runtime.shutdown();
}

File: examples/basic_pool.cpp
#include <cstdint>
#include <cstdio>

#include "wss/scheduler.hpp"
#include "wss/work_stealing_scheduler.hpp"

int main() {
    wss::WorkStealingScheduler scheduler(4);

    auto handle = wss::spawn(scheduler, [] {
        std::uint64_t sum = 0;
        for (std::uint64_t i = 0; i < 1'000'000; ++i) {
            sum += i;
        }
        return sum;
    });

    auto result = handle.join();
    std::printf("result = %llu\n", static_cast<unsigned long long>(result.value()));

    scheduler.shutdown();
}
